src/main/java/uk/gov/hmcts/reform/userprofileapi/repository/IdamRepository.java

```diff
@@ -1,28 +1,42 @@
 package uk.gov.hmcts.reform.userprofileapi.repository;
 
+import com.github.benmanes.caffeine.cache.Cache;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.cache.caffeine.CaffeineCache;
 import org.springframework.stereotype.Component;
 import uk.gov.hmcts.reform.idam.client.IdamClient;
 import uk.gov.hmcts.reform.idam.client.models.UserInfo;
 
+import static java.util.Objects.requireNonNull;
+
 @Component
 @Slf4j
 public class IdamRepository {
 
-    private final IdamClient idamClient;
-
     @Value("${loggingComponentName}")
     private String loggingComponentName;
 
+    private final IdamClient idamClient;
+    private CacheManager cacheManager;
+
     @Autowired
-    public IdamRepository(IdamClient idamClient) {
+    public IdamRepository(IdamClient idamClient, CacheManager cacheManager) {
         this.idamClient = idamClient;
+        this.cacheManager = cacheManager;
     }
 
+    @Cacheable(value = "token")
     public UserInfo getUserInfo(String jwtToken) {
-        log.info("{}:: Inside getUserInfo::", loggingComponentName);
+        CaffeineCache caffeineCache = (CaffeineCache) cacheManager.getCache("token");
+        Cache<Object, Object> nativeCache = requireNonNull(caffeineCache).getNativeCache();
+
+        log.info("{}:: generating Bearer Token, current size of cache: "
```

Comment:
try using `log.info('{}:: generating Bearer Token, current size of cache: {}", loggingComponentName, nativeCache.estimatedSize())`

Reply:
This comment should not be a blocker for merging this PR.

Reply:
Addressed here: https://github.com/hmcts/rd-user-profile-api/pull/725/commits/bd5ae5f1cd143bc92d3d9b29a037201c50418156